modules/ROOT/pages/http/http-listener-ref.adoc

```diff
@@ -1,182 +1,296 @@
+== HTTP Listener Configuration
 
-In response to the request that the Listener receives, you can configure response codes and clarifications for success and error outcomes to present different error messages depending on the outcome.
+To use an HTTP listener, a configuration must be declared with a corresponding connection. This establishes the HTTP server that will listen to requests.
 
-== Setting the Host
+Additionally, a base path can be configured that will apply to all listeners using the configuration.
 
-To deploy a project to CloudHub, setting Host = *0.0.0.0* is required.
+[source,xml,linenums]
+----
+<http:listener-config name="HTTP_Listener_config" basePath="api/v1">
+  <http:listener-connection host="0.0.0.0" port="8081" />
+</http:listener-config>
+----
 
-When you depoly an app locally, set Host = *localhost* for security. Using localhost listens for incoming requests generated inside the server, so the app is not vulnerable to external threats.
+=== HTTP Listener Connection
 
-Setting Host to `0.0.0.0` sets up the Listener to listen on all interfaces of the machine including the following examples:
+The connection defines the host and port where the server will be set up as well as the protocol to use: HTTP for plain connections or HTTPS for TLS connections.
 
-* `+http://localhost:8081/mybasepath+`
-* `+http://127.0.0.1:8081/mybasepath+`
-* `+http://myhost.some.domain:8081/mybasepath+`
+To allow secure connections through HTTPS, a TLS Context should be defined in the connection as well providing a key store for the server. A trust store can also be provided in case
+two way authentication is desired. See the xref:4.1@mule-runtime/tls-configuration.adoc[TLS Configuration] guide for details.
 
-== Setting the Path
+[source,xml,linenums]
+----
+<http:listener-connection host="0.0.0.0" port="8081" protocol="HTTPS">
+  <tls:context>
+    <tls:trust-store path="keystore.jks" password="MyP455W0rD"/>
+  </tls:context>
+</http:listener-connection>
+----
 
-If a request matches the path of multiple Listeners, the Listener having the most specific path has precedence, regardless of the order in which the operations are defined. Only one connector handles each request.
-For example, given the following Listener configurations, the second one receives the request when you use this URL: `account/mulesoft/main-contact`
+[TIP]
+===
+To deploy a project to CloudHub, setting the host to *0.0.0.0* is required since it sets up the connection to listen on all interfaces of the machine.
 
-* Path = `account/(...)`
-* Path = `account/(...)/main-contact`
+When you deploy locally for testing purposes, setting the host to *localhost* is better since it means only requests generated inside the server (your local machine) will be received: the app is not vulnerable to external threats.
```

Comment:
maybe "generated locally: the app is not vulnerable to external threats" instead of "inside the server (your local machine)"

Reply:
I don't understand the point of this. If I'm deploying locally for testing, which threats are you talking about? Isn't this a concern of when you actually deploy to production or to an external staging environment at most?